lib/Data/Chronicle/Writer.pm

```diff
@@ -119,7 +138,10 @@ sub set {
     $value = JSON::to_json($value);
 
     my $key = $category . '::' . $name;
-    $self->cache_writer->set($key => $value, $self->ttl ? ('EX' => $self->ttl) : ());
+    $self->cache_writer->publish($key, $value) if $self->publish_on_set;
```

Comment:
If we'll publish after set, there will be more gap for race: the application (i.e. bom-rpc) might read the updated key before the cache-refresher updated it. 

If we publish event first, the cache-refresher receives it first, and have a better chance to set refreshed value.